api_tests/files/test_views.py

```diff
@@ -207,3 +208,80 @@ def test_must_be_osfstorage(self):
             expect_errors=True,
         )
         assert_equal(res.status_code, 403)
+
+    def test_delete_checked_out_file(self):
+        self.file.provider = 'osfstorage'
+        self.file.save()
+        res = self.app.put_json_api(
+            '/{}files/{}/'.format(API_BASE, self.file._id),
+            {'data': {'id': self.file._id, 'type': 'files', 'attributes': {'checkout': self.user._id}}},
+            auth=self.user.auth,
+        )
+        assert_equal(res.status_code, 200)
+        self.file.reload()
+        with assert_raises(FileNodeorChildCheckedOutError):
+            self.file.delete()
+
+    def test_delete_folder_with_checked_out_file(self):
+        self.app.put_json_api(
+            '/{}files/{}/'.format(API_BASE, self.file._id),
+            {'data': {'id': self.file._id, 'type': 'files', 'attributes': {'checkout': None}}},
+            auth=self.user.auth,
+        )
+        self.file.reload()
+        folder = self.root_node.append_folder('folder')
+        self.file.move_under(folder)
+        self.app.put_json_api(
+            '/{}files/{}/'.format(API_BASE, self.file._id),
+            {'data': {'id': self.file._id, 'type': 'files', 'attributes': {'checkout': self.user._id}}},
+            auth=self.user.auth,
+        )
+        self.file.reload()
+        with assert_raises(FileNodeorChildCheckedOutError):
```

Comment:
You seem to be testing distinct layers of the web stack here: views and model methods. While it's good to test both, they should be tested separately from each other.

In this test, for example, you use HTTP requests to set up your test, but the actual business logic under test (deleting a file that is checked out) is being called as method. 

Better to stay at a consistent level of abstraction:

```python
# views tests

def test_delete_folder_with_checked_out_file(self):
    # test setup ...
   # ... you can use HTTP requests or just set up fixtures/factories directly...
   res = #... send request to delete checked out folder
   # make assertions about the **HTTP response**
   assert_equal(res.status_code, 400)
   #... tests about response body ...
```

```python
# models tests
def test_folder_delete_raises_error_if_child_checked_out(self):
    # ... test set up ...
   with assert_raises(FileNodeCheckedOutError):
        folder.delete()
```